interpreter/interpreter.go

```diff
@@ -136,29 +136,37 @@ func (i *Interpreter) ProcessInit(r *http.Request) error {
 }
 
 func (i *Interpreter) ProcessDeclarations(statements []ast.Statement) error {
-	// Process root declarations and statements
+	// Process root declarations and statements.
+	// Must process backends first because they're referenced by directors.
+	// https://developer.fastly.com/reference/vcl/declarations/
 	for _, stmt := range statements {
-		// Call debugger
+		t, ok := stmt.(*ast.BackendDeclaration)
+		if !ok {
+			continue
+		}
 		i.Debugger.Run(stmt)
+		h := &atomic.Bool{}
+		h.Store(true)
+		// Determine default backend
+		if i.ctx.Backend == nil {
+			i.ctx.Backend = &value.Backend{Value: t, Literal: true, Healthy: h}
+		}
+		if _, ok := i.ctx.Backends[t.Name.Value]; ok {
+			return exception.Runtime(&t.Token, "Backend %s is duplicated", t.Name.Value)
+		}
+		i.ctx.Backends[t.Name.Value] = &value.Backend{Value: t, Literal: true, Healthy: h}
+	}
```

Comment:
![Suggestion](https://img.shields.io/badge/review-suggestion-red)
Could you define this loop as a separated function like `ProcessBackends()` ? it makes be less complex I think.